opslevel/resource_opslevel_filter.go

```diff
@@ -317,28 +367,30 @@ func getConnectiveEnum(connective string) *opslevel.ConnectiveEnum {
 	}
 }
 
-func getFilterPredicates(predicates []filterPredicate) (*[]opslevel.FilterPredicateInput, error) {
+func getFilterPredicates(predicates []filterPredicateModel) (*[]opslevel.FilterPredicateInput, error) {
 	filterPredicateInputs := []opslevel.FilterPredicateInput{}
+
 	for _, predicate := range predicates {
 		tmpPredicateInput := opslevel.FilterPredicateInput{
 			Key:     opslevel.PredicateKeyEnum(predicate.Key.ValueString()),
 			KeyData: predicate.KeyData.ValueStringPointer(),
 			Type:    opslevel.PredicateTypeEnum(predicate.Type.ValueString()),
 			Value:   predicate.Value.ValueStringPointer(),
 		}
+		isCaseSensitiveSet := !predicate.CaseSensitive.IsNull() && !predicate.CaseSensitive.IsUnknown()
+		isCaseInsensitiveSet := !predicate.CaseInsensitive.IsNull() && !predicate.CaseInsensitive.IsUnknown()
 
-		if predicate.CaseSensitive.ValueBool() && predicate.CaseInsensitive.ValueBool() {
+		if isCaseSensitiveSet && isCaseInsensitiveSet {
 			return nil, fmt.Errorf("a predicate should not have 'case_sensitive' and 'case_insensitive' set at the same time")
-		} else if predicate.CaseSensitive.ValueBool() {
-			tmpPredicateInput.CaseSensitive = opslevel.RefOf(true)
-		} else if predicate.CaseInsensitive.ValueBool() {
-			tmpPredicateInput.CaseSensitive = opslevel.RefOf(false)
+		}
+		if isCaseSensitiveSet {
```

Comment:
Replace with `if isCaseSensitiveSet || isCaseInsensitiveSet`, the contents of the `if` are the exact same

Reply:
Almost! The `if` statement has `predicate.CaseSensitive` and the `else if` has `predicate.CaseInsensitive`